apps/jobs/views.py

```diff
@@ -156,6 +145,47 @@ def get_active_jobs_count(serialized_company_data):
         
         return serialized_company_data
 
+    @staticmethod
+    def get_active_jobs_count(serialized_company_data):
+        """
+        Add a new field called "Active Jobs" to the serialized data,
+        that contains the count of active jobs present in the company.
+        """
+
+        for company in serialized_company_data.data:
+            jobs_belong_to_company = Job.objects.filter(
+                company_id=company.get("company_id")
+            )
+            active_jobs = sum(1 for job in jobs_belong_to_company if job.is_active)
+            company.update({"Active Jobs": active_jobs})
+
+        return serialized_company_data
+
+    @action(detail=False, methods=["get"], url_path="details")
+    def retrieve_job_by_id(self, request):
+        """
+        API: /api/v1/jobs/details
+        This method retrieves job data by job_id provided in the query parameter.
+        """
+        job_id = request.query_params.get("job_id")
+
+        if not job_id or not validationClass.is_valid_uuid(job_id):
+            return response.create_response(
+                "Invalid or missing 'job_id' parameter in the query",
+                status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            job_data = Job.objects.get(job_id=job_id)
+        except Job.DoesNotExist:
+            return response.create_response(
+                f"Job with job_id '{job_id}' does not exist", status.HTTP_404_NOT_FOUND
+            )
+
+        serialized_job_data = self.serializer_class(job_data)
+
+        return response.create_response(serialized_job_data.data, status.HTTP_200_OK)
```

Comment:
Do not modify the existing implementation, keep it as it is. 
Here, you have excluded the key "Number of Applicants" from the `retrieve` response, which is crucial in the first place.